test/parallel/test-buffer-alloc.js

```diff
@@ -1024,3 +1024,10 @@ common.expectsError(() => {
   code: 'ERR_INVALID_ARG_VALUE',
   type: TypeError
 });
+
+common.expectsError(() => {
+  Buffer.alloc(1, Buffer.alloc(0));
+}, {
+  code: 'ERR_INVALID_ARG_VALUE',
+  type: TypeError
```

Comment:
This issue seems to be everywhere `ERR_INVALID_ARG_VALUE` is used, so consider this non-blocking, but I'd think `TypeError` is the wrong error to use here. The type of the argument (`Buffer`) is fine. It's the value that's the problem. I think it should be vanilla `Error` or *maybe* `RangeError`.

Reply:
To your point, this PR uses the same error handling logic from #17427. In that PR, [I was against using a `TypeError`](https://github.com/nodejs/node/pull/17427#issuecomment-349354691) for the same reasons you list here, but other collaborators felt differently.

Reply:
I think `TypeError` is fine here.